vnext/Microsoft.ReactNative/Modules/ImageViewManagerModule.cpp

```diff
@@ -147,6 +178,21 @@ void ImageLoader::getSizeWithHeaders(
 }
 
 void ImageLoader::prefetchImage(std::string uri, React::ReactPromise<bool> &&result) noexcept {
+  // VALIDATE URI - file:// abuse PROTECTION (P0 Critical - CVSS 7.8)
+  try {
+    if (uri.find("data:") == 0) {
+      // Validate data URI size to prevent DoS through memory exhaustion
+      ::Microsoft::ReactNative::InputValidation::SizeValidator::ValidateSize(
+          uri.length(), ::Microsoft::ReactNative::InputValidation::SizeValidator::MAX_DATA_URI_SIZE, "Data URI");
+    } else {
+      // Allow http/https only for non-data URIs
+      ::Microsoft::ReactNative::InputValidation::URLValidator::ValidateURL(uri, {"http", "https"});
```

Comment:
This blocks localhost urls, right? During local development we will fetch images from the metro server which will be on localhost. So this will break images when running from metro.